template/next-ts-chakra-ui/pages/index.tsx

```diff
@@ -1,17 +1,28 @@
 import React from 'react';
 import Head from 'next/head';
-import type { NextPage } from 'next';
+import type { NextPage, GetServerSideProps } from 'next';
 
 import { Home } from '@/components/pages';
 import generateMetaTags from '@/utils/meta';
 
-const Index: NextPage = () => {
+type Props = {
+  timestamp: number;
+};
+
+const Index: NextPage<Props> = ({ timestamp }) => {
   return (
     <>
       <Head>{generateMetaTags({ title: 'Home' })}</Head>
-      <Home />
+      <Home timestamp={timestamp} />
     </>
   );
 };
 
+export const getServerSideProps: GetServerSideProps = async () => {
```

Comment:
should this be typed too? `GetServerSideProps<Props>` or what have you?

Reply:
Ah, didn't know you could do that. Fixed.